keepers/validator-keeper/src/mev_commission.rs

```diff
@@ -122,7 +124,7 @@ impl UpdateInstruction for ValidatorMevCommissionEntry {
                 signer: self.signer,
             }
             .to_account_metas(None),
-            data: validator_history::instruction::UpdateMevCommission {}.data(),
+            data: validator_history::instruction::UpdateMevCommission { epoch: self.epoch }.data(),
```

Comment:
Merkle roots for a given epoch are uploaded typically within a few hours after the epoch ends, so to account for this, the keeper will need to submit UpdateMevCommission instructions for the previous epoch after its upload has completed. could achieve this by adding a separate loop in the `main` that submits these at say, the halfway point of the next epoch. Or better yet, continually checks the tip distribution accounts to see which merkle_roots are not None

Reply:
for sure, lmk if what i have works

i added a `update_mev_earned` loop similar to the existing `update_mev_comission` with the main difference that instead of just looking for existing TDAs, it attempts to `TipDistributionAccount::try_deserialize` on the `account.data.data` received from the gMA call and and check for `Some(merkle_root) = tda.merkle_root`

Reply:
perfect